### PR TITLE
GEODE-4691: Gfsh command Start Locator shows inconsistent behavior wi…

### PR DESCRIPTION
**Pull request by sboorlagadda**
…th auto-connecting with a Security Manager

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java [hunk 1]
```diff
@@ -2400,7 +2400,7 @@
   public static final String START_LOCATOR__RUN_MESSAGE = "Starting a Geode Locator in %1$s...";
   public static final String START_LOCATOR__CONNECT = "connect";
   public static final String START_LOCATOR__CONNECT__HELP =
-      "When connect is set to false , Gfsh does not automatically connect to the locator which is started using this command.";
+      "When connect is set to false or when locator is started with a security manager using --J=-Dgemfire.security-manager option, Gfsh does not automatically connect to the locator which is started using this command.";
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: I thought the intent was that the user should see a message like:

`Security Manager is enabled - unable to auto-connect. Please connect with --user and --password` when they start a locator.

Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: I see that the ticket says about updating the help text and I didn't like adding this to help text of `connect` option anyways but I couldn't find a better place in help.

Will update the output showing that he/she is not connected.

Comment (sboorlagadda) [reply geode-core/src/main/java/org/apache/geode/management/internal/cli/i18n/CliStrings.java]: Done fixed both help text and command output as requested in the ticket.

**Comment by sboorlagadda** [comment]
FYI. pre-checkin is green.

File: geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java [hunk 2]
```diff
@@ -375,17 +376,19 @@ private void doAutoConnect(final String locatorHostname, final int locatorPort,
...
     infoResultData.addLine("\n");
-    infoResultData.addLine(CliStrings.format(CliStrings.START_LOCATOR__USE__0__TO__CONNECT,
-        new CommandStringBuilder(CliStrings.CONNECT)
-            .addOption(CliStrings.CONNECT__LOCATOR, locatorHostName + "[" + locatorPort + "]")
-            .toString()));
+    infoResultData
+        .addLine(CliStrings.format(CliStrings.START_LOCATOR__USE__0__TO__CONNECT_WITH_SECURITY,
+            new CommandStringBuilder(CliStrings.CONNECT)
+                .addOption(CliStrings.CONNECT__LOCATOR, locatorHostName + "[" + locatorPort + "]")
+                .addOption(CliStrings.CONNECT__USERNAME).addOption(CliStrings.CONNECT__PASSWORD)
```
Comment (jdeppe-pivotal) [inline geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/StartLocatorCommand.java]: I think this line needs to only be added if `jmxManagerAuthEnabled` is true.

**Pull request merged**